packages/beacon-node/src/chain/archiveStore/interface.ts

```diff
@@ -47,3 +47,13 @@ export interface StateArchiveStrategy {
   maybeArchiveState(finalized: CheckpointWithHex, metrics?: Metrics | null): Promise<void>;
   archiveState(finalized: CheckpointWithHex, metrics?: Metrics | null): Promise<void>;
 }
+
+export interface IArchiveStore {
+  init(): Promise<void>;
+  close(): Promise<void>;
+  scrapeMetrics(): Promise<string>;
+  getHistoricalStateBySlot(
+    slot: number
+  ): Promise<{state: Uint8Array; executionOptimistic: boolean; finalized: boolean} | null>;
+  persistToDisk(): Promise<void>;
```

Comment:
the old BeaconChain has `persistToDisk()` next to `loadFromDisk()` which makes it easy to maintain
here if we go with `init()` which call `loadFromDisk()` we should have `deinit()` which call `persistToDisk()`

also would be nice to have some documentation for these methods

Reply:
I did notice there is also `close()` method. So may be no need `deinit()` but call `persistToDisk()` inside `close()` method?

Reply:
Yes I had persistToDisk inside close before. But later thought to keep minimum change and not replace it's usage. 

I will update the code.

Reply:
@twoeths I checked and found that we have similar pattern for `chain` object as well that `persistToDisk` and `close` are not coupled to each other. And called independently. 

https://github.com/ChainSafe/lodestar/blob/a21ab27b8fe0692f84638c7f5cc802f27d895d58/packages/beacon-node/src/node/nodejs.ts#L332-L333

So I kept that the same pattern in the `archiveStore` as well.